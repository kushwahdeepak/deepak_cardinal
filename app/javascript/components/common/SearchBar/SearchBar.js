import React, { useEffect } from 'react'
import styles from './styles/SearchBar.module.scss'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import FormControl from 'react-bootstrap/FormControl'
import Image from 'react-bootstrap/Image'
import Button from 'react-bootstrap/Button'
import SearchIcon from '../../../../assets/images/talent_page_assets/search-icon.png'

function SearchBar({
    placeholder,
    value,
    onChange,
    onEnterPressed,
    onButtonClick,
    candidateSource,
    candidatePage,
    hideButton=true,
}) {
    return (
      <Row className={`${candidatePage ? styles.candidateSearchBar: styles.atsSearchBar} ${styles.searchBarRow}`}>
        <Col>
          <FormControl
            id={candidatePage ? styles.candidateSearchSpace : ""}
            placeholder={placeholder}
            style={{ textIndent: '40px'}}
            className={candidatePage ? styles.candidateSearch : styles.placeholderText}
            value={value}
            onChange={onChange}
            onKeyPress={(e) => {
              if (e.key === 'Enter') {
                onEnterPressed(e)
              }
            }}
          />

          <div
            style={{
              position: 'absolute',
              left: '30px',
              top: '8px',
              zIndex: '2',
            }}
          >
            <Image src={SearchIcon} style={{filter: candidateSource=='lead_candidate_search' ?  'opacity(0.5)' : ''}} fluid />
          </div>
        </Col>
        { hideButton &&
        <Col xs="auto" className="pl-0">
          <Button className={styles.searchButton} onClick={onButtonClick}>
            Search
          </Button>
        </Col> }
      </Row>
    )
  }

export default SearchBar
