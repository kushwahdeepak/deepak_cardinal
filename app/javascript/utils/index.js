import { act } from 'react-dom/test-utils'

export const findByTestAttr = (wrapper, attr) => {
    return wrapper.find(`[data-test='${attr}']`)
}

export const timeoutResolve = (time = 0, response = {}) =>
    new Promise((resolve) => {
        setTimeout(() => resolve(response), time)
    })

export const wait = (time = 0) => {
    const timer = timeoutResolve(time)

    return act(async () => {
        await timer
    })
}
export function emailIsValid(email) {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function generateRandom(min, max) {
    return Math.floor(min + Math.random() * (max + 1 - min))
}

export function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
}

export function createMarkedList(array) {
    if (!Array.isArray(array)) {
        return []
    }
    return array.map((candidate) => ({
        id: candidate.id,
        name: candidate.id,
        fullName: `${candidate.first_name} ${candidate.last_name}`,
        emailAddress: candidate.email_address,
        isChecked: false,
    }))
}

export function stringReplace(string,toSymbol,fromSymbol){
    return string.replace(toSymbol, fromSymbol)
}

export function capitalize(string) {
    return string.replace(/\b(\w)/g, s => s.toUpperCase());
}

export function skillIsNotEmpty(value){
    return value.trim().length > 0;
}

export function firstCharacterCapital(value){
    return value?.charAt(0)?.toUpperCase() + value?.slice(1)
}
