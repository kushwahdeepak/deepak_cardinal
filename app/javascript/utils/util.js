import isNil from 'lodash.isnil'
import isEmpty from 'lodash.isempty'

class Util {
    /**
     * Define stage values
     */
    static STAGES = ['first_interview', 'second_interview', 'offer', 'reject']

    /**
     * Tests whether a single object attribute includes the at
     * least one string from the given set.
     */
    static attributePassesStringSetFilter(object, attr, stringSet) {
        if (isNil(object[attr])) return false

        // We "OR" the tag tests: the attribute needs to contain only one tag
        return stringSet
            .map((string) => string.trim().toLowerCase())
            .reduce(
                (passes, string) =>
                    passes ||
                    Util.attributeToString(object[attr])
                        .toLowerCase()
                        .includes(string),
                false
            )
    }

    /**
     * Tests whether at least one of a set of object attributes
     * includes every word from a given piece of text.
     */
    static objectPassesTextFilter(object, attrs, text) {
        text = text.trim().toLowerCase()

        if (text === '') return true

        const words = text.split(' ')

        // Join all the given attribute names to lowercase, stringified attribute values
        const allAttrsString = attrs
            .filter((attr) => !isNil(object[attr]))
            .map((attr) => Util.attributeToString(object[attr]).toLowerCase())
            .join('')

        // Is every word included in the allAttrsString?
        return words.reduce(
            (matchesAll, word) => matchesAll && allAttrsString.includes(word),
            true
        )
    }

    static attributeToString(attr) {
        if (typeof attr === 'string') {
            return attr
        } else if (Array.isArray(attr)) {
            return attr.join(',')
        } else if (typeof attr === 'object') {
            return JSON.stringify(attr)
        } else {
            return '' + attr
        }
    }

    static numberWithCommas(x) {
        return x?.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
    }

    /**
     * Returns number of results in format [x] - [y] of [z]
     * @param {number} totalJobsCount Total number of jobs
     * @param {number} pageCount Total number of pages
     * @param {number} activePage Currently active page
     * @param {number} pageSize How many results to display per page
     */
    static displayNumberOfResults(
        totalJobsCount,
        pageCount,
        activePage,
        pageSize,
        totalPersonCount=null,
        candidatePage=false,
        showdisplaycount=true
    ) {
        if (totalJobsCount == 0) return '0 - 0 of 0'
        const offset = pageSize * pageCount - totalJobsCount
        const lastJobIndex =
            (activePage + 1) * pageSize > totalJobsCount
                ? totalJobsCount
                : (activePage + 1) * pageSize
        const firstJobIndex =
            activePage == 0
                ? 1
                : activePage == pageCount - 1
                ? lastJobIndex - pageSize + offset + 1
                : lastJobIndex - pageSize + 1
        return `${firstJobIndex} - ${lastJobIndex} of ${Util.numberWithCommas(
            (candidatePage && showdisplaycount) ? totalPersonCount : totalJobsCount

        )}`
    }

    /**
     * Returns a string indicating the candidate's title and current
     * company, based on available information on the Person object
     * @param {object} candidate Person object from our db
     */
    static candidateCompanyString(candidate) {
        const position = candidate.company_position
        const currentCompany = candidate.current_company

        if (isEmpty(position) && isEmpty(currentCompany)) return ''

        if (!isEmpty(position) && !isEmpty(currentCompany))
            return `${position} at ${currentCompany}`

        if (!isEmpty(position)) return position

        if (!isEmpty(currentCompany)) return `at ${currentCompany}`
    }

    /**
     * Converts the given array of objects with ids to an object whose
     * keys are the ids of the array objects.
     *
     * Every object in the given array must have a unique 'id' property defined.
     *
     * @param {Array} array
     * @returns {Object}
     */
    static arrayToObject(array) {
        const obj = {}
        array.forEach((element) => (obj[element.id] = element))
        return obj
    }

    /**
     * Makes a copy of the given array and replace the value at 'index' with 'value'
     *
     * @param {Array} array
     * @param {Number} index
     * @param {*} value
     * @returns {Array}
     */
    static replaceInArrayCopy_i(array, index, value) {
        const clone = array.slice()
        clone[index] = value
        return clone
    }

    /**
     * Makes a copy of the given array and replaces the element with the same
     * .id property as obj.id with obj
     *
     * @param {Array} array
     * @param {Object} obj
     * @returns {Array}
     */
    static replaceInArrayCopy(array, obj) {
        const index = array.findIndex((element) => element.id === obj.id)
        return Util.replaceInArrayCopy_i(array, index, obj)
    }

    static deleteInArrayCopy(array, index) {
        const copy = array.slice()
        copy.splice(index, 1)

        return copy
    }

    static handleUndefined(value){
        if (value === undefined || value === 'undefined')
        return '';
        else
        return value
    }

    static handleUndefinedFullName(firstName, lastName){
      if(firstName === "undefined" || firstName === undefined)
      return lastName
      else if(lastName === "undefined" || lastName === undefined)
      return firstName
      else
      return firstName + ' ' + lastName     }

    static capitalize(value){
         return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
    }
}

export default Util
