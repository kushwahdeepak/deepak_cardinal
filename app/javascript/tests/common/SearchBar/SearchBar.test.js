import React from 'react';
import SearchBar from '../../../components/common/SearchBar/SearchBar';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr } from '../../../utils';

configure({adapter: new Adapter()});

const initialShallowRender = (props = {}) => {
	return shallow(<SearchBar {...props}/>);
};

const initialMountRender = (props = {}) => {
	return mount(<SearchBar {...props}/>);
};

describe('SearchBar component',  () => {
	let wrapper;
	let fullWrapper;

	beforeEach(() => {

		const props = {
		};
		wrapper = initialShallowRender(props);
		fullWrapper = initialMountRender(props);
	});

	afterEach(() => {
		fullWrapper.unmount();
	});

	it('should render SearchBar component', () => {
		expect(wrapper.exists()).toBe(true);
	});

	it('should match with a snapshot', () => {
		const component = renderer.create(<SearchBar />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});
