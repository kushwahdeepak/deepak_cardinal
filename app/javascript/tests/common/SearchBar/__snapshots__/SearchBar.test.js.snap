// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should match with a snapshot 1`] = `
<div>
  <div
    className="top-search-bar-target"
  >
    <div
      style={
        Object {
          "height": "1px",
          "margin": "0 auto",
          "marginBottom": "20px",
        }
      }
    />
    <div
      style={
        Object {
          "marginBottom": "20px",
          "width": "100%",
        }
      }
    >
      <img
        alt="Logo White"
        height="50"
        src="test-file-stub"
        style={
          Object {
            "display": "block",
            "margin": "0 auto",
          }
        }
        width="50"
      />
    </div>
    <h1
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      World-Class Talent On Demand
    </h1>
    <h2
      style={
        Object {
          "color": "white",
          "fontWeight": 350,
          "textAlign": "center",
        }
      }
    >
      30,000+ Top Talent Who Are Open to New Opportunities Now
    </h2>
  </div>
  <div>
    <div
      className="input-group"
    >
      <div
        className="bg-white input-group-prepend"
      >
        <img
          className="rounded"
          src="test-file-stub"
        />
      </div>
      <input
        className="border-0 form-control"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search here.."
        type="text"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Find Candidate
        </button>
      </div>
    </div>
  </div>
</div>
`;
